internal/commands/app/create_inputs.go

```diff
@@ -156,56 +166,117 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
-func (i *createInputs) resolveCluster(client atlas.Client, groupID string) (dataSourceCluster, error) {
+func (i *createInputs) resolveClusters(ui terminal.UI, client atlas.Client, groupID string) ([]dataSourceCluster, error) {
 	clusters, err := client.Clusters(groupID)
 	if err != nil {
-		return dataSourceCluster{}, err
+		return nil, err
 	}
-	var clusterName string
-	for _, cluster := range clusters {
-		if i.Cluster == cluster.Name {
-			clusterName = cluster.Name
-			break
-		}
+
+	existingClusters := map[string]interface{}{}
+	for _, c := range clusters {
+		existingClusters[c.Name] = c
 	}
-	if clusterName == "" {
-		return dataSourceCluster{}, errors.New("failed to find Atlas cluster")
+	nonExistingClusters := make([]string, 0, len(i.Clusters))
+
+	dsClusters := make([]dataSourceCluster, 0, len(i.Clusters))
+	for idx, clusterName := range i.Clusters {
+		if _, ok := existingClusters[clusterName]; !ok {
+			nonExistingClusters = append(nonExistingClusters, clusterName)
+			continue
+		}
+
+		serviceName := clusterName
+		if len(i.ClusterServiceNames) > idx {
+			serviceName = i.ClusterServiceNames[idx]
+		} else {
+			if !ui.AutoConfirm() {
+				if err := ui.AskOne(&serviceName, &survey.Input{Message: fmt.Sprintf("Enter a Service Name for Cluster '%s'", clusterName), Default: serviceName}); err != nil {
+					return nil, err
+				}
+			}
+		}
+
+		dsClusters = append(dsClusters,
+			dataSourceCluster{
+				Name: serviceName,
+				Type: realm.ClusterType,
+				Config: configCluster{
+					ClusterName:         clusterName,
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			})
 	}
-	dsCluster := dataSourceCluster{
-		Name: "mongodb-atlas",
-		Type: "mongodb-atlas",
-		Config: configCluster{
-			ClusterName:         clusterName,
-			ReadPreference:      "primary",
-			WireProtocolEnabled: false,
-		},
+
+	if len(nonExistingClusters) > 0 {
+		ui.Print(terminal.NewWarningLog("Please note, the following Atlas clusters '%s' were not linked because they could not be found", strings.Join(nonExistingClusters[:], ", ")))
```

Comment:
ok so this is almost there, just a couple of notes:

* start the warning message as: `Note: The following Atlas clusters...` (no "Please", capital "The")
* let's invert the structure a bit so the "dynamic" portion of the message always is at the end
  * `Note: The following Atlas clusters were not linked because they could not be found: %s`
  * we don't need to wrap the whole list of separated cluster names with `''`
* I'd like to move this Warning and Confirm outside the scope of this function and to the command `Handler`
  * We should instead just add `nonExistingClusters` (a `[]string`) to the return signature of this function
  * We can do the same for `nonExistingDatalakes`
  * The warning message can then account for both `clusters` and `datalakes`, with a single confirm for the user, before potentially proceeding
* in the case of `!proceed`, we don't need to return an error, just `nil` is fine